### PR TITLE
Add ping performance test

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] improves usage of the locust.io perfomance tool
- [x] created an aggressive test for pinging deployments

the test simulates 100 users calling 10x the root of the osparc platform and 1x the /v0/health endpoint for 1 minute.
the test fails if the average response time is above 200ms and if the failure rate is above 1%

my current statistics
- local: 0% failure, the average response time: 44ms with a max at 1098ms, doing about 2122 requests/seconds in total.
- master: 0% failure, the average response time: 79ms with a max at 2124ms, doing about 1198 requests/seconds in total.

### CODE REVIEW

File: tests/performance/Makefile [hunk 1]
```diff
@@ -16,42 +20,29 @@ __check_defined = \
...
 
-.PHONY: up
-up:
+.PHONY: up down run
+up: ## runs distributed osparc locust using target=locust_test_file.py, optional host can be set
 	@$(call check_defined, target, please define target file when calling $@ - e.g. ```make $@ target=MY_LOCUST_FILE.py```)
-	@export TARGET=$(target); \
-	docker-compose --file docker-compose.yml up
+	@export LOCUST_FILE=$(target); \
+	export TARGET_HOST=$(if $(host),$(host),"http://$(get_my_ip):9081"); \
```
Comment (pcrespov) [inline tests/performance/Makefile:36]: MINOR: TARGET_HOST  is really TARGET_URL?

File: tests/performance/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Usage in CI (headless mode)
+
+```console
+# builds the distributed locust worker/master image
+make build
+# runs the test defined by locust_files/platform_ping_test.py going to http://127.0.0.1:8089 allows to see the UI
+make test target=platform_ping_test.py
+# removes the containers
+make down
```
Comment (GitHK) [inline tests/performance/README.md:28]: I do not see the difference with the above `# Usage` paragraph

**Review by GitHK — APPROVED** [review]

File: tests/performance/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,9 @@
-FROM locustio/locust:1.4.4
+FROM locustio/locust:1.5.3
 
+COPY ./requirements /build/requirements
```
Comment (pcrespov) [inline tests/performance/Dockerfile:3]: I think hyou forgot to commit requirements folder?
Got
```
=> ERROR [2/3] COPY ./requirements /build/requirements  
```

**Review by pcrespov — CHANGES_REQUESTED** [review]

File: tests/performance/Dockerfile (hunk 3 above)
Comment (pcrespov) [reply tests/performance/Dockerfile:3]: Or probably forgot to remove the line ... since you list the reqs below

Comment (sanderegg) [reply tests/performance/Dockerfile:3]: no I forgot to remove that line. thanks good catch!

File: tests/performance/README.md (hunk 2 above)
Comment (sanderegg) [reply tests/performance/README.md:28]: ```make up``` vs ```make test```

File: tests/performance/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+```
+
+## Usage in CI (headless mode)
+
+```console
+# builds the distributed locust worker/master image
+make build
+# runs the test defined by locust_files/platform_ping_test.py going to http://127.0.0.1:8089 allows to see the UI
```
Comment (pcrespov) [inline tests/performance/README.md:25]: I guess the comment in 25 is wrong .. since it is headless?

File: tests/performance/README.md (hunk 2 above)
Comment (pcrespov) [reply tests/performance/README.md:28]: Above is **up** and below is **test**

File: tests/performance/Makefile (hunk 1 above)
Comment (sanderegg) [reply tests/performance/Makefile:36]: done

**Review by pcrespov — APPROVED** [review]

File: tests/performance/README.md (hunk 4 above)
Comment (sanderegg) [reply tests/performance/README.md:25]: corrected

**Pull request merged**